images/metering-ansible-operator/roles/meteringconfig/library/k8s_event.py

```diff
@@ -0,0 +1,382 @@
+        argument_spec.update(EVENT_ARG_SPEC)
+        return argument_spec
+
+    def execute_module(self):
+        """ Module execution """
+        self.client = self.get_api_client()  # mostly auth
+
+        api_version = "v1"
+        name_args = self.params.get("name")
+        namespace_args = self.params.get("namespace")
+        message = self.params.get("message")
+        reason = self.params.get("reason")
+        reporting_component = self.params.get("reportingComponent")
+        event_type = self.params.get("type")
+        source = self.params.get("source")
+
+        definition = defaultdict(defaultdict)
+
+        def_meta = definition["metadata"]
+        def_meta["name"] = self.params.get("name")
+        def_meta["namespace"] = self.params.get("namespace")
+
+        def_involvedObject = definition["involvedObject"]
+        def_involvedObject["namespace"] = self.params.get("namespace")
+        def_involvedObject["apiVersion"] = self.params.get("apiVersion")
+        def_involvedObject["kind"] = self.params.get("kind")
+        def_involvedObject["name"] = self.params.get("name")
+        def_involvedObject["uid"] = self.params.get("uid")
+        def_involvedObject["resourceVersion"] = self.params.get("resourceVersion")
+
+        resource = self.find_resource(
+            "Event", "v1", fail=True
+        )  # finds resource or gets it from the client api, looks for resrouce
+        # second call to find resource for involved object, set kind to arg of involved OBject kind
+        # rnewesource = self.find_resource('involvedObjectKind', 'v1', fail=True)
+
+        # this just establishes tha vars and seperates out the reason and count from returned cli object
+        priorCount = 1
+        try:
+            priorEvent = resource.get(
+                name=def_meta["name"],
+                namespace=def_meta["namespace"]
+            )
+            priorReason = priorEvent["reason"]
+            print(priorReason)
+            print("current reason is %s" % reason)
+            priorCount = priorEvent["count"]
+            print("the count from the prior event is %i" % priorCount)
+
+            if priorEvent is not None and priorEvent["reason"] != reason:
+                print("If reason changed %i" % priorCount)
+                now = datetime.datetime.now()
+                rfc = kubernetes.config.dateutil.format_rfc3339(now)
+                #firstTimestamp = rfc
+                priorCount = 1
+                firstTimestamp = rfc
+                print("the value of the firstTimestamp new event is", firstTimestamp)
+            else:
+                # priorCount = 1
+                priorCount +=1
+                print("the reason has changed", priorCount)
+        except openshift.dynamic.exceptions.NotFoundError:
+            pass
+
+        involvedObject_resourceVersion = "1"
+        involvedObject_uid = "1"
+        try:
+            totalEvent = resource.get(
```

Comment:
Your querying the wrong resource here. Your querying the event, not the involved object. You need to use self.find_resource to be able to query the right type of resource for the involvedObject.

For example:

```
involvedObjectResource = self.find_resource(involvedObject["kind"], "v1", fail=True)
# if the "kind" for involvedObject exists
if involvedObjectResource:
    try:
        # get the involvedObject if it exists
        apiInvolvedObject = involvedObjectResource.get(name=involvedObject["name"], namespace=involvedObject["namespace"])
        # sets the involvedObject uid and resourceVersion from the values if the object exists
        # retrieved from the API
        involvedObject["uid"] = apiInvolvedObject["metadata"]["uid"]
        involvedObject["resourceVersion"] = apiInvolvedObject["metadata"]["resourceVersion"]
    except openshift.dynamic.exceptions.NotFoundError:
        # involvedObject doesn't exist, so simply do not set uid/resourceVersion on it
        pass
```